src/search/landmarks/landmark_count_heuristic.cc

```diff
@@ -199,30 +201,19 @@ int LandmarkCountHeuristic::compute_heuristic(const State &ancestor_state) {
     int h = get_heuristic_value(ancestor_state);
 
     if (use_preferred_operators) {
-        BitsetView landmark_info = lm_status_manager->get_reached_landmarks(ancestor_state);
-        LandmarkNodeSet reached_lms = convert_to_landmark_set(landmark_info);
-        generate_helpful_actions(state, reached_lms);
+        generate_helpful_actions(state);
     }
 
     return h;
 }
 
-bool LandmarkCountHeuristic::check_node_orders_disobeyed(const LandmarkNode &node,
-                                                         const LandmarkNodeSet &reached) const {
-    for (const auto &parent : node.parents) {
-        if (reached.count(parent.first) == 0) {
-            return true;
-        }
-    }
-    return false;
-}
-
-bool LandmarkCountHeuristic::generate_helpful_actions(
-    const State &state, const LandmarkNodeSet &reached) {
-    /* Find actions that achieve new landmark leaves. If no such action exist,
-     return false. If a simple landmark can be achieved, return only operators
-     that achieve simple landmarks, else return operators that achieve
-     disjunctive landmarks */
+bool LandmarkCountHeuristic::generate_helpful_actions(const State &state) {
+    /*
+      Find actions that achieve new landmark leaves. If no such action exist,
+      return false. If a simple landmark can be achieved, return only operators
+      that achieve simple landmarks, else return operators that achieve
+      disjunctive landmarks.
+    */
```

Comment:
This comment contradicts the change log message. The change log message suggests that we are happy with achieving *any* landmark. But here the first sentence says "landmark leaves", which means landmarks whose parents have all been accepted.

Reply:
Another error in the comment: it implies the false dichotomy that landmarks are either simple or disjunctive; there is the third option of being conjunctive. From the comment, it is not clear how conjunctive landmarks are handled. (I assume this is because this code predates the addition of conjunctive landmarks.)

Reply:
Actually, I wonder how the code handles conjunctive landmarks. From looking at this a bit more, I think it either ignores them or handles them wrong. Can we discuss this?